apps/etl-func/src/domain/message-status.ts

```diff
@@ -0,0 +1,37 @@
+import * as z from "zod";
+
+const rejectionReasonSchema = z
+  .enum(["SERVICE_NOT_ALLOWED", "USER_NOT_FOUND", "UNKNOWN"])
+  .default("UNKNOWN");
+
+const rejectedMessageStatusSchema = z.object({
+  fiscalCode: z.string().min(1).optional(),
+  isArchived: z.boolean().default(false),
+  isRead: z.boolean().default(false),
+  messageId: z.string().min(1),
+  rejection_reason: rejectionReasonSchema,
+  status: z.literal("REJECTED"),
+  updatedAt: z.number(),
+});
+
+const notRejectedMessageStatusValueEnum = z.enum([
+  "ACCEPTED",
+  "THROTTLED",
+  "FAILED",
+  "PROCESSED",
+]);
+
+const notRejectedMessageStatusSchema = z.object({
+  fiscalCode: z.string().min(1).optional(),
+  isArchived: z.boolean().default(false),
+  isRead: z.boolean().default(false),
+  messageId: z.string().min(1),
+  status: notRejectedMessageStatusValueEnum,
+  updatedAt: z.number(),
+});
```

Comment:
```suggestion
const statusEnum = z.enum([
  "ACCEPTED",
  "THROTTLED",
  "FAILED",
  "PROCESSED",
  "REJECTED",
]);

const rejectedMessageStatusSchema = z.object({
  fiscalCode: z.string().min(1).optional(),
  isArchived: z.boolean().default(false),
  isRead: z.boolean().default(false),
  messageId: z.string().min(1),
  rejection_reason: rejectionReasonSchema,
  status: statusEnum.extract(["REJECTED"]),
  updatedAt: z.number(),
});

const notRejectedMessageStatusSchema = z.object({
  fiscalCode: z.string().min(1).optional(),
  isArchived: z.boolean().default(false),
  isRead: z.boolean().default(false),
  messageId: z.string().min(1),
  status: statusEnum.exclude(["REJECTED"]),
  updatedAt: z.number(),
});
```

Why don't we have an enum that has all the status and then use zod to choose which one to use in the objects?

Reply:
You are right, take a look here [45f9d40](https://github.com/pagopa/io-messages/pull/59/commits/45f9d40a83e65f22d3aa7f0864e59a9fab1238e1)